File: src/Entity/Participation.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

use App\Repository\ParticipationRepository;
use App\Entity\User;

#[ORM\Entity(repositoryClass: ParticipationRepository::class)]
#[ORM\Table(name: 'participation')]
class Participation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $idParticipation = null;

    public function getIdParticipation(): ?int
    {
        return $this->idParticipation;
    }

    public function setIdParticipation(int $idParticipation): self
    {
        $this->idParticipation = $idParticipation;
        return $this;
    }

    #[ORM\ManyToOne(targetEntity: Hackathon::class, inversedBy: 'participations')]
    #[ORM\JoinColumn(name: 'id_hackathon', referencedColumnName: 'id_hackathon')]
    private ?Hackathon $hackathon = null;

    public function getHackathon(): ?Hackathon
    {
        return $this->hackathon;
    }

    public function setHackathon(?Hackathon $hackathon): self
    {
        $this->hackathon = $hackathon;
        return $this;
    }


    #[ORM\ManyToOne(targetEntity: User::class, inversedBy: 'participations')]
    #[ORM\JoinColumn(name: 'id_participant', referencedColumnName: 'idUser')]
    private ?User $participant = null;

    public function getParticipant(): ?User
    {
        return $this->participant;
    }

    public function setParticipant(?User $participant): self
    {
        $this->participant = $participant;
        return $this;
    }

    #[ORM\Column(type: 'datetime', nullable: false)]
    private ?\DateTimeInterface $dateInscription = null;

    public function getDateInscription(): ?\DateTimeInterface
    {
        return $this->dateInscription;
    }

    public function setDateInscription(\DateTimeInterface $dateInscription): self
    {
        $this->dateInscription = $dateInscription;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: false)]
    private ?string $statut = null;

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(string $statut): self
    {
        $this->statut = $statut;
        return $this;
    }

    




}

File: OneDrive/Bureau/Hackify-integrationn/src/main/java/models/Participation.java
package models;

import java.time.LocalDateTime;

public class Participation {
    private int idParticipation;
    private int idParticipant;
    private int id_hackathon;
    private LocalDateTime dateInscription;
    private boolean estEquipe;
    private String statut;

    public Participation(){

    }
    public Participation( int idHackathon, int idParticipant , LocalDateTime dateInscription, String statut) {
        this.id_hackathon = idHackathon;
        this.idParticipant = idParticipant;
        this.dateInscription = LocalDateTime.now();
        this.statut = statut;
    }
    public Participation(int idParticipation, int idHackathon, int idParticipant , LocalDateTime dateInscription, String statut) {
        this.idParticipation=idParticipation;
        this.id_hackathon = idHackathon;
        this.idParticipant = idParticipant;
        this.dateInscription = LocalDateTime.now();
        this.statut = statut;
    }
    public Participation(int idHackathon , int idParticipant) {
        this.id_hackathon=idHackathon;
        this.idParticipant = idParticipant;
    }
    public Participation(int idHackathon) {
        this.id_hackathon=idHackathon;

    }

    public int getIdParticipation() {
        return idParticipation;
    }

    public int getIdHackathon() {
        return id_hackathon;
    }

    public int getIdParticipant() {
        return idParticipant;
    }

    public String getStatut() {
        return statut;
    }

    public LocalDateTime getDateInscription() {
        return dateInscription;
    }


    public void setStatut(String statut) {
        this.statut = statut;
    }


    @Override
    public String toString() {
        return "Participation{" +
                "idParticipation=" + idParticipation +
                ", idParticipant=" + idParticipant +
                ", id_hackathon=" + id_hackathon +
                ", statut='" + statut + '\'' +
                ", dateInscription=" + dateInscription +
                '}';
    }
}
